### PR TITLE
persist accounts

### PR DESCRIPTION
**Pull request by butonic**
- partially impelments the new api(list, get, create, delete users) 
- persists accounts as json
- indexes them using bleve
- [ ] autocreate default accounts if the accounts dir does not exist yet

replaces https://github.com/owncloud/ocis-accounts/pull/30

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/provider/tokenizer.go  1
- pkg/provider/bleve.go  24
- pkg/service/v0/service.go  9
         

Clones added
============
- pkg/provider/bleve.go  2
- pkg/service/v0/service.go  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5707362&bid=18679832)

File: pkg/service/v0/service.go [hunk 1]
```diff
@@ -2,73 +2,311 @@ package service
...
+	if err != nil {
+		panic(err)
+	}
+	f, err := os.Open(filepath.Join(cfg.Server.AccountsDataPath, "accounts"))
+	if err != nil {
+		log.Error().Err(err).Str("dir", filepath.Join(cfg.Server.AccountsDataPath, "accounts")).Msg("could not open acconts folder")
+		panic(err)
+	}
+	list, err := f.Readdir(-1)
+	f.Close()
```
Comment (C0rby) [inline pkg/service/v0/service.go:55]: Shouldn't this be `defer f.Close()`?

File: pkg/service/v0/service.go [hunk 2]
```diff
@@ -2,73 +2,311 @@ package service
...
 
 // New returns a new instance of Service
 func New(cfg *config.Config) Service {
-	s := Service{
-		Config: cfg,
+	// read all user and group records
+
+	// for now recreate index on every start
+	os.RemoveAll(filepath.Join(cfg.Server.AccountsDataPath, "index.bleve"))
+	os.MkdirAll(filepath.Join(cfg.Server.AccountsDataPath, "accounts"), 0700)
```
Comment (refs) [inline pkg/service/v0/service.go:41]: `0644`? I don't see why the user needs execute permissions

Comment (refs) [inline pkg/service/v0/service.go:41]: I know it's tedious, but error checks?

File: pkg/service/v0/service.go (hunk 1 above)
Comment (refs) [reply pkg/service/v0/service.go:55]: nah no need for `defer` here. The file descriptor is not used beyond this point nor on a goroutine, so the early it gets freed the better

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/provider/tokenizer.go  1
- pkg/provider/bleve.go  24
- pkg/service/v0/service.go  10
         

Clones added
============
- pkg/provider/bleve.go  2
- pkg/service/v0/service.go  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5707362&bid=18679832)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/provider/bleve.go  24
- pkg/service/v0/service.go  10
- pkg/provider/tokenizer.go  1
         

Clones added
============
- pkg/provider/bleve.go  2
- pkg/service/v0/service.go  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5707362&bid=18679832)

File: pkg/service/v0/service.go (hunk 2 above)
Comment (butonic) [reply pkg/service/v0/service.go:41]: 0700 because it is a dir

**Pull request merged**